src/NextGen/DonationForm/Rules/SubscriptionFrequencyRule.php

```diff
@@ -0,0 +1,41 @@
+<?php
+namespace Give\NextGen\DonationForm\Rules;
+
+use Closure;
+use Give\Donations\ValueObjects\DonationType;
+use Give\Vendors\StellarWP\Validation\Contracts\ValidatesOnFrontEnd;
+use Give\Vendors\StellarWP\Validation\Contracts\ValidationRule;
+
+class SubscriptionFrequencyRule implements ValidationRule, ValidatesOnFrontEnd {
+
+    /**
+     * @unreleased
+     */
+    public static function id(): string
+    {
+        return 'subscriptionFrequency';
+    }
+
+    /**
+     * @unreleased
+     */
+    public static function fromString(string $options = null): ValidationRule
+    {
+        return new self();
+    }
+
+    /**
+     * @unreleased
+     */
+    public function __invoke($value, Closure $fail, string $key, array $values): bool
+    {
+        $donationType = new DonationType($values['donationType']);
+
+        return $donationType->isSubscription() && is_numeric($value);
```

Comment:
The second condition refers to `is_numeric($value)`, but `$value` (singular) doesn't seem to be defined. An array `$values` (plural) is an argument of the method - do you mean to validate that each value is numeric?

Reply:
@kjohnson i'm not sure what you mean by:
>`$value` (singular) doesn't seem to be defined

The rule closure passes the following params: `($value, Closure $fail, string $key, array $values)`

Reply:
Oh, ok, I see the first `$value` parameter now. So then, what is the last argument?

Reply:
The last argument is array of values